# File: my_costmap_params.yaml
global_frame: "map"
robot_base_frame: "base_link"
update_frequency: 10
publish_frequency: 10
transform_tolerance: 0.3
robot_radius: 1.0
always_send_full_costmap: true

resolution: 0.05              # meters/cell
width: 50.0                   # total width in meters (1000 cells * 0.05 m/cell)
height: 50.0                  # total height in meters
origin_x: -25.0               # bottom-left corner x-coordinate
origin_y: -25.0               # bottom-left corner y-coordinate
origin_z: 0.0                 # typically 0 for 2D

# Plugins (layers)
plugins:
  - {name: static_layer,    type: "costmap_2d::StaticLayer"}
  - {name: obstacle_layer,  type: "costmap_2d::ObstacleLayer"}
  - {name: inflation_layer, type: "costmap_2d::InflationLayer"}

# Static Layer
static_layer:
  enabled: true
  map_topic: "/map"            # Ensure /map is being published correctly
  subscribe_to_updates: true  # Set to true if /map is dynamic

obstacle_layer:
  enabled: true  # Enable the obstacle layer
  observation_sources: laser_scan cone_lines_scan  # List all observation sources

  # Laser scan configuration
  laser_scan:
    data_type: LaserScan
    topic: "/scan"  # Topic for laser scan data
    marking: true  # Mark obstacles detected by laser scan
    clearing: true  # Allow clearing of obstacles when laser scan no longer detects them

  # Cone lines configuration
  cone_lines_scan:
    data_type: PointCloud2
    topic: "/cone_lines_obstacles"  # Topic for cone line obstacles
    marking: true  # Mark obstacles detected by cone lines
    clearing: false  # Do not clear cone line obstacles automatically
    obstacle_range: 100


# Inflation Layer
inflation_layer:
  enabled: true
  cost_scaling_factor: 0
  inflation_radius: 1.2    # meters
